moveit_core/robot_state/test/robot_state_test.cpp

```diff
@@ -553,6 +554,72 @@ TEST(FK, OneRobot)
   EXPECT_NEAR_TRACED(state.getGlobalLinkTransform("link_e").translation(), Eigen::Vector3d(2.8, 0.6, 0));
 }
 
+class LoadPR2 : public testing::Test
+{
+protected:
+  virtual void SetUp()
+  {
+    std::string res_path(MOVEIT_TEST_RESOURCES_DIR);
+
+    urdf_model = urdf::parseURDFFile(res_path + "/pr2_description/urdf/robot.xml");
+    srdf_model.reset(new srdf::Model());
+    srdf_model->initFile(*urdf_model, res_path + "/pr2_description/srdf/robot.xml");
+    robot_model.reset(new moveit::core::RobotModel(urdf_model, srdf_model));
+  }
+
+  virtual void TearDown()
+  {
+  }
+
+protected:
+  urdf::ModelInterfaceSharedPtr urdf_model;
+  srdf::ModelSharedPtr srdf_model;
+  moveit::core::RobotModelConstPtr robot_model;
+};
+
+void generateTestTraj(std::vector<std::shared_ptr<robot_state::RobotState>>& traj,
+                      const moveit::core::RobotModelConstPtr& robot_model,
+                      const robot_model::JointModelGroup* joint_model_group)
+{
+  std::size_t n_joints = joint_model_group->getJointModelNames().size();
+  std::vector<double> joint_positions(n_joints, 0.0);
+  // 3 waypoints with zero joints
+  for (std::size_t traj_ix = 0; traj_ix < 3; ++traj_ix)
+  {
+    std::shared_ptr<robot_state::RobotState> robot_state(new robot_state::RobotState(robot_model));
+    robot_state->setJointGroupPositions(joint_model_group, joint_positions);
+    traj.push_back(robot_state);
+  }
+  // 4th waypoint with a large jump in first joint
+  joint_positions[0] = 1.01;
+  std::shared_ptr<robot_state::RobotState> robot_state(new robot_state::RobotState(robot_model));
+  robot_state->setJointGroupPositions(joint_model_group, joint_positions);
+  traj.push_back(robot_state);
+}
+
+TEST_F(LoadPR2, testJointSpaceJumpAbsolute)
+{
+  const robot_model::JointModelGroup* joint_model_group = robot_model->getJointModelGroup("right_arm");
+  std::vector<std::shared_ptr<robot_state::RobotState>> traj;
+  generateTestTraj(traj, robot_model, joint_model_group);
+  ASSERT_EQ(traj.size(), 4);  // traj should have 4 waypoints, with a large jump in the last point
+
+  double fraction = robot_state::RobotState::testJointSpaceJump(joint_model_group, traj, 1.0, 1.0);
+  EXPECT_EQ(traj.size(), 3);  // traj should be cut
+  EXPECT_NEAR(fraction, 3. / 4., 0.01);
+}
+
+TEST_F(LoadPR2, testJointSpaceJumpRelative)
+{
+  const robot_model::JointModelGroup* joint_model_group = robot_model->getJointModelGroup("right_arm");
+  std::vector<std::shared_ptr<robot_state::RobotState>> traj;
+  generateTestTraj(traj, robot_model, joint_model_group);
+
+  double fraction = robot_state::RobotState::testJointSpaceJump(joint_model_group, traj, 1.0);
+  EXPECT_EQ(traj.size(), 3);  // traj should be cut
+  EXPECT_NEAR(fraction, 3. / 4., 0.01);
+}
+
```

Comment:
What about adding tests that ensure that `testJointSpaceJump` correctly relays to `testJointSpaceJumpRelative` and `testJointSpaceJumpAbsolute`?

Reply:
I'll add that